### PR TITLE
[@storybook__*] Delete Storybook types

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

Select one of these and delete the others:

If removing a declaration:
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

---
All packages were migrated to TS (which is tracked in storybookjs/storybook#5030) and provide their own typings.

I deleted the typings for the following packages:
- `@storybook/addon-actions`
- `@storybook/addon-backgrounds`
- `@storybook/addon-centered`
- `@storybook/addon-jest`
- `@storybook/addon-knobs`
- `@storybook/addon-links`
- `@storybook/addon-options`
- `@storybook/addon-viewport`
- `@storybook/addons`
- `@storybook/channels`
- `@storybook/html`
- `@storybook/preact`
  Set to `5.2.1` since `5.2` typings were already published here even tho that package version wasn't released yet
- `@storybook/react`
- `@storybook/react-native`
- `@storybook/vue`

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]

**Review by joscha — COMMENTED** [review]
What happens to consumers of pre-5.2.0 versions if there are any problems with the types? Adding the packages to `notNeededPackages.json` looks good to me, but I think we should keep the typings around for a little longer?

**Review by joscha — CHANGES_REQUESTED** [review]
Requesting changes until the question is addressed, slo it doesn't accidentally get merged.

**Review by joscha — CHANGES_REQUESTED** [review]
.

**Comment by MichaelDeBoey** [comment]
@joscha I just followed the workflow as explained in the PR template and the [docs](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#removing-a-package).

Other deleted types (see #34111 & #37358) just deleted the corresponding folders too.

**Comment by joscha** [comment]
> @joscha I just followed the workflow as explained in the PR template and the [docs](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#removing-a-package) [](/DefinitelyTyped/DefinitelyTyped/blob/HEAD@{2019-09-18T00:57:21Z}/README.md#removing-a-package).

ah, so the output of this PR is from `npm run not-needed -- typingsPackageName asOfVersion sourceRepoURL [libraryName].`? I guess that's expected then? Feels a bit strange that we wouldn't be able to update typings for older versions any more, but 🤷‍♂

File: notNeededPackages.json
Comment (ExE-Boss) [inline notNeededPackages.json]: I’d prefer if this was:
```suggestion
            "asOfVersion": "5.2.0"
```
But I don’t know how that would affect `types‑publisher`.

Comment (MichaelDeBoey) [reply notNeededPackages.json]: Like I said in my PR description:
> Set to `5.2.1` since `5.2` typings were already published here even tho that package version wasn't released yet

**Comment by MichaelDeBoey** [comment]
@joscha Yes it does 🙂

**Comment by MichaelDeBoey** [comment]
Tests fail because of `@storybook/react` isn't whitelisted, but since microsoft/types-publisher#658 is merged it should be.

Don't know how I can fix this tbh. 😕

Somebody who can help me out? 🤔

CC @joscha @jicjjang @HyunSeob @adhrinae @kiyopikko @halfmatthalfcat @martynaskadisa @amacleay @MLoughry @alanhchoi @azmenak @jessepinho @simonhn @gaetanmaisse @adam187 @Vinnl @bmatcuk @ChristianMurphy @9renpoto @ddayguerrero @wapgear @dandean @alechill @iRoachie @ceyhuno @pntgupta @jurgisrudaks @yoyoys

**Comment by ExE-Boss** [comment]
The build needs to be re‑ran now that https://github.com/microsoft/types-publisher/pull/658 has been merged.

**Comment by MichaelDeBoey** [comment]
@ExE-Boss Is there something I can do to make it re-run? 🤔
My tests won't run locally either. 😕

**Comment by ExE-Boss** [comment]
You need someone with repository write access to restart the build on Travis:
<table><td>Travis Restart Build
<br/><sup>Picture taken on one of my own repositories.</sup>
</td></table>

---

Alternatively, you can just run:
```pwsh
git commit --amend --no-edit
git push -f
```

**Comment by MichaelDeBoey** [comment]
@ExE-Boss Don't think that  will work, since `yarn test` is failing locally too.

**Comment by MichaelDeBoey** [comment]
@ExE-Boss Like I thought: still failing because `@storybook/react` isn't whitelisted yet 😕

**Comment by MichaelDeBoey** [comment]
Still get the following error:

```sh
TS1259: Module '"/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/react/index"' can only be default-imported using the 'esModuleInterop' flag
```

But can't figure out how to fix it.

Can somebody point me in the right direction please? 🤔

CC @joscha @jicjjang @HyunSeob @adhrinae @kiyopikko @halfmatthalfcat @martynaskadisa @amacleay @MLoughry @alanhchoi @azmenak @jessepinho @simonhn @gaetanmaisse @adam187 @Vinnl @bmatcuk @ChristianMurphy @9renpoto @ddayguerrero @wapgear @dandean @alechill @iRoachie @ceyhuno @pntgupta @jurgisrudaks @yoyoys @ExE-Boss

**Comment by MichaelDeBoey** [comment]
Still get the following error:

```sh
TS1259: Module '"/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/react/index"' can only be default-imported using the 'esModuleInterop' flag
```

But can't figure out how to fix it.

Can somebody point me in the right direction please? 🤔

CC @joscha @jicjjang @HyunSeob @adhrinae @kiyopikko @halfmatthalfcat @martynaskadisa @amacleay @MLoughry @alanhchoi @azmenak @jessepinho @simonhn @gaetanmaisse @adam187 @Vinnl @bmatcuk @ChristianMurphy @9renpoto @ddayguerrero @wapgear @dandean @alechill @iRoachie @ceyhuno @pntgupta @jurgisrudaks @yoyoys @ExE-Boss

**Comment by ExE-Boss** [comment]
@MichaelDeBoey

You’re supposed to use:
```ts
import react = require('react');
```

when dealing with older, CommonJS‑style modules with a top‑level export.

## See also:
- https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Comment by MichaelDeBoey** [comment]
@ExE-Boss Can you provide some more info, so I can make an issue in their repo? 🤔

You can do it yourself too if you want 🙂

**Comment by MichaelDeBoey** [comment]
@ExE-Boss Can you provide some more info, so I can make an issue in their repo? 🤔

You can do it yourself too if you want 🙂

**Comment by MichaelDeBoey** [comment]
Or maybe somebody from the @storybookjs team can help me out? 🤔

CC @gaetanmaisse @kroeder @ndelangen @shilman @emilio-martinez @lonyele @benoitdion @dandean @sairus2k

**Comment by MichaelDeBoey** [comment]
I've reverted the removal  of the `@storybook/react`, so this can already be merged.

I'll  create a new PR for it when this one's merged. 🙂

**Comment by ndelangen** [comment]
What can I do to help here?

**Comment by MichaelDeBoey** [comment]
@ndelangen I'll tag you in the new PR after this one's merged.
But that one will have the same issue as we had here, so maybe you can have a look at the error that we had? 🤔

https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/591252348

**Comment by MichaelDeBoey** [comment]
Ping @joscha

**Comment by joscha** [comment]
Sorry, I still don't really have an answer as to what happens with the pre-5.2.0 versions after we merge this, so I don't really know whether this should be approved or not.

**Comment by MichaelDeBoey** [comment]
@joscha This is just the normal flow of deleting typings from the repo.

Since their's a new version published to `npm` when we update typings, I guess this won't happen anymore and the old `@types/storybook__*` will still be existing on the `npm` registry.

**Comment by joscha** [comment]
Yeah, sorry, let me clarify. I understand we don't need to worry about the
new version, I also understand removing them makes sense. I am just
wondering what happens if someone uses 4.x for a while longer (not out of
the question as the upgrade is not straightforward) and we encounter a
typing error - what do we do then?

**Comment by MichaelDeBoey** [comment]
@joscha I think this is a general thing that needs to be discussed elsewhere, no? 🤔

**Comment by ndelangen** [comment]
@joscha We don't actively maintain or support 4.x anymore.

I'm personally not too concerned with this "what happens if" scenario, users have options:

My recommendation would be to use a fork, a local type definition or https://www.npmjs.com/package/patch-package

And hopefully the option of upgrading is available..

My main concern is that storybook 5.x users should remove the packages as soon as possible, because they are incompatible with the 5.x releases.

**Comment by joscha** [comment]
I do understand, we already migrated to v5 so I am not being cautious
because of myself, I sincerely think the v4 -> v5 update we did was not the
easiest so I am a bit worried about us just dropping these types and making
it our user's problem.
But I can see you are all pushing for this and if noone else thinks it is a
problem, maybe I shouldn't either...

**Review by joscha — APPROVED** [review]

**Comment by Mussabaheen** [comment]
GREAT

**Pull request merged**
